seed/seedwriter/writer.go

```diff
@@ -1225,9 +1223,67 @@ func (w *Writer) Downloaded(fetchAsserts AssertsFetchFunc) (complete bool, err e
 	if err := w.ensureARefs(nil, fetchAsserts); err != nil {
 		return false, err
 	}
+
+	if err := w.checkPrereqs(); err != nil {
+		return false, err
+	}
+
 	return true, nil
 }
 
+func (w *Writer) checkPrereqs() error {
+	modes := make([]string, 0, len(w.availableByMode))
+	modes = append(modes, "run")
+	fixed := 1
+	if _, ok := w.availableByMode["ephemeral"]; ok {
+		modes = append(modes, "ephemeral")
+		fixed = 2
+	}
+	for m := range w.availableByMode {
+		if m == "run" || m == "ephemeral" {
+			continue
+		}
+		modes = append(modes, m)
+	}
+	sort.Strings(modes[fixed:])
```

Comment:
given we stop on first error the idea is to have a fixed order so that we fail at the same point always, and we start from run that should usually have the most snaps... I will add a comment